File: Modeling and computing/P1.java
import java.util.Scanner;

public class P1 {

    // Function to read a 2D array of doubles from stdin
    public static double[][] readDoubleMatrixFromStdin() {
        Scanner sc = new Scanner(System.in);

        // Read the first line as an integer (number of lines to follow)
        int n = sc.nextInt();
        sc.nextLine();  // Consume leftover newline

        // Prepare a 2D array to hold n rows
        double[][] data = new double[n][];

        // Read each of the next n lines
        for (int i = 0; i < n; i++) {
            String line = sc.nextLine().trim();

            // Split the line by spaces
            String[] tokens = line.split("\\s+");

            // Convert to double
            double[] row = new double[tokens.length];
            for (int j = 0; j < tokens.length; j++) {
                row[j] = Double.parseDouble(tokens[j]);
            }

            // Store this row in the 2D array
            data[i] = row;
        }

        sc.close();
        return data;
    }

    public static void main(String[] args) {
        // Read the matrix from stdin
        double[][] data = readDoubleMatrixFromStdin();
        // for loop to iterate over each student
        for (double[] student : data) {
            int participation = (int) student[0];
            int homework = (int) student[1];
            int homework_sufficient = (int) student[2];
            double exam = student[3];
            double project = student[4];

            double final_grade = (0.7 * exam) + (0.3 * project);
            // passing conditions
            if (
                exam >= 5.5 
                && project >= 5.5 
                && final_grade >= 5.5 
                && participation >= 8
                && homework >= 3
                && (homework >= 5 || (3 <= homework && homework < 5  && (homework_sufficient == 1 || homework_sufficient == 2)))
            ) {

                System.out.printf("PASS %.1f%n", final_grade);
            } else {
                System.out.println("FAIL");
            }
        }
    }
}

File: Modeling and computing/P1.py
import sys

def compute_final_grade(x, p):
    final_grade = (0.7 * x) + (0.3 * p)
    return final_grade

def main():
    # Read all input at once
    input_lines = sys.stdin.read().splitlines()
    n = int(input_lines[0])  # First line is the number of students

    for i in range(1, n + 1):  # Iterate over each student's data
        # Parse the input line
        data = input_lines[i].split()
        participation = int(data[0])  # Participation must be an integer
        homework_correct = int(data[1])  # Homework correct must be an integer
        homework_sufficient = int(data[2])  # Homework sufficient must be an integer
        exam = float(data[3])  # Exam score is a float
        project = float(data[4])  # Project score is a float

        # Compute the final grade
        final_grade = compute_final_grade(exam, project)



        # Check if the student passes
        is_pass = (
            exam >= 5.5 and
            project >= 5.5 and
            final_grade >= 5.5 and
            participation >= 8 and
            (homework_correct >= 5 or (3 <= homework_correct < 5 and homework_sufficient in [1, 2]))
        )

        # Output the result
        if is_pass:
            print("PASS", round(final_grade, 1))
        else:
            print("FAIL")

    # Flush the output buffer to ensure all output is displayed
    sys.stdout.flush()

if __name__ == "__main__":
    main()
    